Cubical/Foundations/RelationalStructure.agda

```diff
@@ -45,60 +39,132 @@ InducedQuotientStr : (S : Type ℓ → Type ℓ') (ρ : StrRel S ℓ'')
+quotientPropRel R .fst a t = [ a ] ≡ t
+quotientPropRel R .snd _ _ = squash/ _ _
+
+record QERDescends (S : Type ℓ → Type ℓ') (ρ : StrRel S ℓ'')
+  (A B : TypeWithStr ℓ S) (R : QuasiEquivRel (typ A) (typ B) ℓ) : Type (ℓ-max ℓ' ℓ'')
   where
   private
-    module E = Bisim→Equiv R
+    module E = QER→Equiv R
 
   field
     quoᴸ : InducedQuotientStr S ρ A E.Rᴸ
     quoᴿ : InducedQuotientStr S ρ B E.Rᴿ
-    path : PathP (λ i → S (ua E.Thm i)) (quoᴸ .fst) (quoᴿ .fst)
-
-open BisimDescends
-
--- A notion of structured relations is standard when
--- (a) Given a structured type A and equivalence relation R, there is at most one quotient structure on A/R
--- (b) Any bisimulation descends to the quotients if and only if it is structured
-record isUnivalentRel (S : SetStructure ℓ ℓ') (ρ : StrRel (S .struct) ℓ'') : Type (ℓ-max (ℓ-max (ℓ-suc ℓ) ℓ') ℓ'')
+    rel : ρ .rel (graphRel (E.Thm .fst)) (quoᴸ .fst) (quoᴿ .fst)
+
+open QERDescends
+open isQuasiEquivRel
+
+structuredQER→structuredEquiv : {S : Type ℓ → Type ℓ'} (ρ : StrRel S ℓ'')
+  (θ : SuitableStrRel S ρ)
+  (A B : TypeWithStr ℓ S) (R : QuasiEquivRel (typ A) (typ B) ℓ)
+  → ρ .rel (R .fst .fst) (A .snd) (B .snd)
+  → QERDescends S ρ A B R
+structuredQER→structuredEquiv ρ θ (X , s) (Y , t) R r .quoᴸ =
+  θ .quo (X , s) (QER→EquivRel R)
+    (subst (λ R' → ρ .rel R' s s) correction
+      (θ .transitive (R .fst) (invPropRel (R .fst)) r (θ .symmetric (R .fst) r)))
+    .fst
   where
-  field
-    propQuo : {A : TypeWithStr ℓ (S .struct)}
-      (R : Σ[ R ∈ (typ A → typ A → Type ℓ) ] isEquivRel R)
-      → isProp (InducedQuotientStr (S .struct) ρ A (R .fst))
-    descends : {A B : TypeWithStr ℓ (S .struct)}
-      (R : Bisimulation (typ A) (typ B) ℓ)
-      → (ρ .rel (R .fst) (A .snd) (B .snd) → BisimDescends (S .struct) ρ A B R)
-      × (BisimDescends (S .struct) ρ A B R → ρ .rel (R .fst) (A .snd) (B .snd))
-
-  descendsEquiv : {A B : TypeWithStr ℓ (S .struct)}
-    (R : Bisimulation (typ A) (typ B) ℓ)
-    → ρ .rel (R .fst) (A .snd) (B .snd) ≃ BisimDescends (S .struct) ρ A B R
-  descendsEquiv R =
-    isPropEquiv→Equiv
-      (ρ .prop (R .snd .isBisimulation.prop) _ _)
-      isPropDescends
-      (descends R .fst)
-      (descends R .snd)
-    where
-    module E = Bisim→Equiv R
-
-    isPropDescends : (d₀ d₁ : BisimDescends (S .struct) ρ _ _ R) → d₀ ≡ d₁
-    isPropDescends d₀ d₁ j .quoᴸ = propQuo (bisim→EquivRel R) (d₀ .quoᴸ) (d₁ .quoᴸ) j
-    isPropDescends d₀ d₁ j .quoᴿ = propQuo (bisim→EquivRel (invBisim R)) (d₀ .quoᴿ) (d₁ .quoᴿ) j
-    isPropDescends d₀ d₁ j .path =
-      isProp→PathP
-        {B = λ j →
-          PathP (λ i → S .struct (ua E.Thm i))
-          (isPropDescends d₀ d₁ j .quoᴸ .fst)
-          (isPropDescends d₀ d₁ j .quoᴿ .fst)}
-        (λ i → isOfHLevelPathP' 1 (S .set squash/) _ _)
-        (d₀ .path) (d₁ .path)
-        j
-
+  correction : compPropRel (R .fst) (invPropRel (R .fst)) .fst ≡ QER→EquivRel R .fst .fst
+  correction =
+    funExt₂ λ x₀ x₁ → ua
```

Comment:
ua of isPropEquiv->Equiv is hPropExt: https://github.com/agda/cubical/blob/master/Cubical/Foundations/Univalence.agda#L42